### PR TITLE
Batch submit task

### PR DESCRIPTION
**Pull request by ahuang11**
Adds https://github.com/PrefectHQ/prefect-aws/issues/38

### CODE REVIEW

File: tests/test_batch.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+from moto import mock_batch, mock_iam, mock_s3
+from prefect import flow
+
+from prefect_aws.batch import batch_submit
+
+
+@pytest.fixture(scope="function")
+def s3_client(aws_credentials):
+    with mock_s3():
+        yield boto3.client("s3", region_name="us-east-1")
```
Comment (desertaxle) [inline tests/test_batch.py:14]: Looks like this fixture isn't used.

**Review by desertaxle — CHANGES_REQUESTED** [review]
`batch_submit` looks good. I think it's important to also have the waiting functionality that's mentioned in #38. Can you also add a `client_waiter` task in this PR or in a follow up PR?

**Comment by ahuang11** [comment]
Perhaps another PR, and I think @mtconger97 might be interested in helping out with it!

File: mkdocs.yml [hunk 2]
```diff
@@ -37,3 +37,4 @@ nav:
     - Credentials: credentials.md
     - S3: s3.md
     - Secrets Manager: secrets_manager.md
+    - Batch: batch.md
```
Comment (desertaxle) [inline mkdocs.yml:40]: I think you need to create this file

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: tests/conftest.py [hunk 3]
```diff
@@ -1,16 +1,9 @@
...
 
 from prefect_aws import AwsCredentials
 from prefect_aws.client_parameters import AwsClientParameters
 
 
-@pytest.fixture(scope="session", autouse=True)
-def prefect_db():
-    with prefect_test_harness():
-        yield
-
```
Comment (desertaxle) [inline tests/conftest.py:12]: Why did you remove this fixture?

Comment (ahuang11) [reply tests/conftest.py:12]: I think it was crashing tests earlier (e.g. `from prefect.utilities.testing import ...`), but now I think it's not used since the tests are still passing?

Comment (ahuang11) [reply tests/conftest.py:12]:
```
asyncio: mode=auto
collected 58 items

tests/test_batch.py .                                                                                                                                                                              [  1%]
tests/test_client_parameters.py ....                                                                                                                                                               [  8%]
tests/test_s3.py ...............                                                                                                                                                                   [ 34%]
tests/test_secrets_manager.py ......................................                                                                                                                               [100%]

===================================================================================== 58 passed in 103.01s (0:01:43) =====================================================================================
```

Comment (desertaxle) [reply tests/conftest.py:12]: Based on the comments in #27 where it was added, it's a nice thing to have to spin up a temporary DB per test. I think we should leave it in and correct the import path (`from prefect.test.utilities import prefect_test_harness`).

**Review by desertaxle — APPROVED** [review]
